app/models/payment_gateway_setting.rb

```diff
@@ -18,9 +18,11 @@ class PaymentGatewaySetting < ApplicationRecord
   # In our DB :test is stored for some providers
 
   def symbolized_settings
-    keys = PaymentGateway::GATEWAYS.map(&:fields).map(&:keys).flatten.uniq
+    settings = gateway_settings
+    return settings.symbolize_keys if settings.is_a? Hash
 
-    gateway_settings.permit(keys).to_h.symbolize_keys.except(:test)
+    keys = PaymentGateway.all.map(&:fields).map(&:keys).flatten.uniq
+    settings.permit(keys).to_h.symbolize_keys.except(:test)
```

Comment:
On line 18 there is a comment `In our DB :test is stored for some providers`. So perhaps we need to keep `:test`? I'm not sure, just a question.

Reply:
I don't know either but I'd say let's keep the logic as it was (with .except)

Reply:
Didn't notice that old version was same :)